x-pack/plugins/monitoring/public/application/hooks/use_charts.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import moment from 'moment';
+import { useContext, useState, useEffect, useRef } from 'react';
+import { useHistory } from 'react-router-dom';
+import { useKibana } from '../../../../../../src/plugins/kibana_react/public';
+import { MonitoringTimeContainer } from '../hooks/use_monitoring_time';
+
+export function useCharts() {
+  const { services } = useKibana<{ data: any }>();
+  const history = useHistory();
+  const { handleTimeChange } = useContext(MonitoringTimeContainer.Context);
+
+  const [zoomInLevel, setZoomInLevel] = useState(0);
+  const [onBrushHappened, _setOnBrushHappened] = useState(false);
+
+  const onBrushHappenedRef = useRef(onBrushHappened);
+
+  const setOnBrushHappened = (data: boolean) => {
+    onBrushHappenedRef.current = data;
+    _setOnBrushHappened(data);
+  };
+
+  useEffect(() => {
+    const popstateHandler = () => {
+      if (onBrushHappenedRef.current) {
+        setOnBrushHappened(false);
+      } else {
```

Comment:
The pop state event is also fired when the onBrush event is fired, and I couldn't find out why this happens. So I added a flag to know when the onBrush event is fired to prevent changing the zoom level at the wrong moment. 
In Angular, this was handled by removing the listener before updating the state and adding it again after some milliseconds, but the same trick didn't seem to work in React.

Reply:
Interesting. Might be good to capture this in a code comment.